### PR TITLE
Kube api rules atuomated tests

### PR DESCRIPTION
**Pull request by oren-zohar**
- adds missing kube-api automated tests
- refactor kube-api rules testing environment 
____

fixes: https://github.com/elastic/security-team/issues/3874

### CODE REVIEW

File: tests/product/tests/test_kube_api_rules.py [hunk 1]
```diff
@@ -7,46 +7,86 @@
 import pytest
 
 from commonlib.utils import get_evaluation, get_resource_identifier
-from product.tests.kube_rules import cis_5_1_5
+from product.tests.kube_rules import *
+
+
+def todict(obj):
```
Comment (gurevichdmitry) [inline tests/product/tests/test_kube_api_rules.py:13]: @oren-zohar, 1. Where is this function used? 2. May be a part of utils.

**Review by gurevichdmitry — COMMENTED** [review]
@oren-zohar, you changed basic functions used for deploying resources to k8s cluster. Kube-api rules tests works with mock-pod.yaml. Since create / delete from yaml is used also for start/stop agent/cloudbeat, please ensure that this functionality also working as expected.

**Comment by oren-zohar** [comment]
> @oren-zohar, you changed basic functions used for deploying resources to k8s cluster. Kube-api rules tests work with mock-pod.yaml. Since create / delete from yaml is used also for start/stop agent/cloudbeat, please ensure that this functionality also working as expected.

how can I test them?

**Comment by gurevichdmitry** [comment]
> > @oren-zohar, you changed basic functions used for deploying resources to k8s cluster. Kube-api rules tests work with mock-pod.yaml. Since create / delete from yaml is used also for start/stop agent/cloudbeat, please ensure that this functionality also working as expected.
> 
> how can I test them?

Just need to verify that functions start_agent and delete_from_yaml in kubernetes.py are working. The usage is in product tests folder conftest.py file -> data fixture.

**Comment by oren-zohar** [comment]
@gurevichdmitry the `data` fixture is working of course, since the `test_env` fixture depends on it, and it is used in the kube rules tests

**Review by gurevichdmitry — APPROVED** [review]

**Comment by DaveSys911** [comment]
@oren-zohar  I think u need to either get someone from @elastic/cloudbeat group to approve or remove it.
Since  me and dima aren't in it.

**Review by amirbenun — APPROVED** [review]

**Review by DaveSys911 — APPROVED** [review]
LGTM

**Pull request merged**